tests/test_versions.py

```
@@ -280,7 +305,241 @@ def test_build_version_details_data_formats_run_exports_from_py_rattler() -> Non
+    loader = VersionDataLoader(client=cast(Client, object()))
+    record = _make_repo_data_record(name="demo")
+    preview_key = ("demo", "1.2.3", "py313h123_0", 0, "noarch", record.file_name)
+
+    async def _fake_get_package_paths(
+        _preview_key: tuple[str, str, str, int, str, str], _url: str
+    ) -> list[PackageFile]:
+        raise RuntimeError("paths.json missing")
+
+    monkeypatch.setattr(loader, "get_package_paths", _fake_get_package_paths)
+
+    with pytest.raises(RuntimeError, match="paths.json missing"):
```

Ruff (0.15.10)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)